src/Dapr.Actors/Runtime/ActorRuntime.cs

```diff
@@ -12,54 +12,43 @@ namespace Dapr.Actors.Runtime
     using System.Text.Json;
     using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
 
     /// <summary>
     /// Contains methods to register actor types. Registering the types allows the runtime to create instances of the actor.
     /// </summary>
     public sealed class ActorRuntime
     {
-        private const string TraceType = "ActorRuntime";
-
         // Map of ActorType --> ActorManager.
         private readonly Dictionary<string, ActorManager> actorManagers = new Dictionary<string, ActorManager>();
 
         private ActorSettings actorSettings;
 
-        internal ActorRuntime()
+        private readonly ILogger<ActorRuntime> logger;
+
+        internal ActorRuntime(DaprActorRuntimeOptions options, ILoggerFactory loggerFactory)
         {
             this.actorSettings = new ActorSettings();
+            this.logger = loggerFactory.CreateLogger<ActorRuntime>();
+            DaprInteractor = new DaprHttpInteractor(loggerFactory:loggerFactory);
+
+            // Create ActorManagers, override existing entry if registered again.
+            foreach(var actorServiceFunc in options.actorServicesFunc)
+            {
+                var actorServiceFactory = actorServiceFunc.Value ?? ((type) => new ActorService(type, loggerFactory));
+                var actorService = actorServiceFactory.Invoke(actorServiceFunc.Key);
+
+                this.actorManagers[actorServiceFunc.Key.ActorTypeName] = new ActorManager(actorService, loggerFactory);
+            }
         }
 
         /// <summary>
         /// Gets actor type names registered with the runtime.
         /// </summary>
         public IEnumerable<string> RegisteredActorTypes => this.actorManagers.Keys;
 
-        internal static IDaprInteractor DaprInteractor => new DaprHttpInteractor();
-
-        /// <summary>
-        /// Registers an actor with the runtime.
-        /// </summary>
-        /// <typeparam name="TActor">Type of actor.</typeparam>
-        /// <param name="actorServiceFactory">An optional delegate to create actor service. This can be used for dependency injection into actors.</param>
-        public void RegisterActor<TActor>(Func<ActorTypeInformation, ActorService> actorServiceFactory = null)
-            where TActor : Actor
-        {
-            var actorTypeInfo = ActorTypeInformation.Get(typeof(TActor));
-
-            ActorService actorService;
-            if (actorServiceFactory != null)
-            {
-                actorService = actorServiceFactory.Invoke(actorTypeInfo);
-            }
-            else
-            {
-                actorService = new ActorService(actorTypeInfo);
-            }
+        internal static IDaprInteractor DaprInteractor;
```

Comment:
Did you come up with a plan to fix this?

If you want to limit the scope/size of this PR it might be good to leave this as-is for now and revisit in a separate change. We could also review and maybe decide to eliminate some of the logging in that class, which might remove the need to refactor.

Reply:
let's discuss offline. Would want to keep the PR scope limited though.

Reply:
fixed as discussed